src/term.rs

```diff
@@ -157,19 +176,38 @@ impl Term {
         })
     }
 
+    /// Return a terminal for the given Read/Write pair.
+    #[cfg(unix)]
+    pub fn read_write_pair<R, W>(read: R, write: W) -> Term
+    where
+        R: io::Read + std::fmt::Debug + AsRawFd + Send + 'static,
+        W: Write + std::fmt::Debug + AsRawFd + Send + 'static,
+    {
+        Term::with_inner(TermInner {
+            target: TermTarget::ReadWritePair(ReadWritePair {
+                read: Arc::new(Mutex::new(read)),
+                write: Arc::new(Mutex::new(write)),
+            }),
+            buffer: None,
+        })
+    }
+
     /// Returns the style for the term
     #[inline]
     pub fn style(&self) -> Style {
-        match self.target() {
+        match self.inner.target {
             TermTarget::Stderr => Style::new().for_stderr(),
             TermTarget::Stdout => Style::new().for_stdout(),
+            // TODO: Needs an implementation. But also: is not Copy.
+            #[cfg(unix)]
+            TermTarget::ReadWritePair(_) => Style::new().for_stderr(),
```

Comment:
I don't think this is good. What if people want to emulate stdout?

Reply:
@stuhood Any update on this PR?

Reply:
Sorry for the delay: will try to get back to this this week.

Reply:
@pksunkara : Would you be ok with changing the signature of this method to return `&Style` so that we can avoid creating a new `Style` instance each time it is called? If so, I think that we can make the `Style` configurable in the constructor.

Reply:
Hmm, no. I think the pattern people use is that the style is movable.

Reply:
`Style` does not currently derive `Copy`, so I don't think that that is the case... I could derive `Copy` there, and then each call to this method would clone the `Style`: is that preferable?

Reply:
Yup.

Reply:
Added an optional `Style`.